pkg/network/common/egress_dns.go

```diff
@@ -133,7 +133,9 @@ func (e *EgressDNS) Sync() {
 		case response := <-e.dnsResponse:
 			if response.Changed {
 				// TODO: Should this be non-blocking?
+				start := time.Now()
 				e.Updates <- e.getEgressDNSUpdates(response.Name)
+				warnSlowOperation("Writing to egressDNS Updates channel", start)
```

Comment:
You could also rewrite this to use a non-blocking channel that's just used to signal the presence of updates, and a `GetUpdates()` method to get the actual changes. Like in #139.

Reply:
Note to myself: Not  outdated TODO

Reply:
Now this is done on a separate goroutine, technically it's not the same thing but it achieves the same.